packages/synapse-react-client/src/components/Plot/SynapsePlot.tsx

```diff
@@ -1,135 +1,164 @@
+  const isHorizontal = toBoolean(horizontal)
+  const isDisplayModebar = toBoolean(displayModebar)
+  const config: Partial<Plotly.Config> = {
+    displayModeBar: isDisplayModebar,
+  }
+  const layout: Partial<Plotly.Layout> = {
+    showlegend: isShowLegend,
+    title,
+    barmode: barmode
+      ? (barmode.toLowerCase() as 'stack' | 'group' | 'overlay' | 'relative')
+      : undefined,
+  }
+  if (xtitle) {
+    layout.xaxis = {
+      title: xtitle,
     }
-    if (xaxistype) {
-      layout.xaxis = {
-        ...layout.xaxis,
-        xaxistype: xaxistype.toLowerCase(),
-      }
+  }
+  if (xaxistype) {
+    layout.xaxis = {
+      ...layout.xaxis,
+      type: xaxistype.toLowerCase() as AxisType,
     }
-    if (ytitle) {
-      layout.yaxis = {
-        title: ytitle,
-      }
+  }
+  if (ytitle) {
+    layout.yaxis = {
+      title: ytitle,
     }
-    // init plot_data
-    const plotData: any = []
-    const orientation = isHorizontal ? 'v' : 'h'
-    const headers = queryData.queryData.queryResult?.queryResults.headers ?? []
-    for (let i = 0; i < headers.length - 1; i += 1) {
-      // make an entry for each set of data points
-      plotData[i] = {
-        orientation,
-        name: headers[i + 1].name,
-        type: type.toLowerCase(),
-        x: [],
-        y: [],
-      }
+  }
+  // init plot_data
+  const plotData: Partial<Plotly.PlotData>[] = []
+  const orientation = isHorizontal ? 'h' : 'v'
+  const headers = queryData.queryResult?.queryResults.headers ?? []
+  for (let i = 0; i < headers.length - 1; i += 1) {
+    // make an entry for each set of data points
+    plotData[i] = {
+      orientation,
+      name: headers[i + 1].name,
+      type: type.toLowerCase() as PlotType,
+      x: [],
+      y: [],
+      customdata: [],
     }
-    // grab all the data
-    for (const row of queryData.queryData.queryResult?.queryResults.rows ??
-      []) {
-      for (let j = 1; j < row.values.length; j += 1) {
-        // create pairs of data
-        const rowValues: any = row.values
-        plotData[j - 1].x.push(rowValues[0])
-        plotData[j - 1].y.push(rowValues[j])
-      }
+  }
+  // grab all the data
+  for (const row of queryData.queryResult?.queryResults.rows ?? []) {
+    for (let j = 1; j < row.values.length; j += 1) {
+      // create pairs of data
+      const rowValues = row.values
+      const xArray = plotData[j - 1]!.x as Plotly.Datum[]
+      const yArray = plotData[j - 1]!.y as Plotly.Datum[]
+      const customdata = plotData[j - 1]!.customdata as Plotly.Datum[]
+      xArray.push(rowValues[0])
+      yArray.push(rowValues[j])
+      customdata.push(JSON.stringify(row))
     }
-    return <Plot layout={layout} data={plotData} />
   }
 
-  public render() {
-    if (!this.state.isLoaded) {
-      return null
+  let onPlotClick:
+    | ((event: Readonly<Plotly.PlotMouseEvent>) => void)
+    | undefined
+  if (onCustomPlotClick && queryContext) {
+    onPlotClick = eventData => {
```

Comment:
On plot point click, pass the selected Row as well as the entire query context that was passed to us (from the query  wrapper plot nav)!